site/app/controllers/forum/ForumController.php

```diff
@@ -632,6 +632,33 @@ public function bookmarkThread(bool $type) {
         return $this->core->getOutput()->renderJsonSuccess($response);
     }
 
+    /**
+     * @Route("/courses/{_semester}/{_course}/forum/threads/unread", methods={"POST"})
+     * @return JsonResponse
+     */
+    public function markThreadUnread() {
+        $thread_id = $_POST["thread_id"];
+        $current_user = $this->core->getUser()->getId();
+        $this->core->getQueries()->unreadThread($current_user, $thread_id);
+        $response = ['user' => $current_user, 'thread' => $thread_id];
+        return JsonResponse::getSuccessResponse($response);
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/forum/posts/unread", methods={"POST"})
+     * @return JsonResponse
+     */
+    public function markPostUnread() {
+        $thread_id = $_POST["thread_id"];
+        $last_viewed_timestamp = $_POST["last_viewed_timestamp"];
+        // format the last viewed timestamp to be in the same format as the database
+        $last_viewed_timestamp = DateUtils::parseDateTime($last_viewed_timestamp, $this->core->getUser()->getUsableTimeZone())->format("Y-m-d H:i:sO");
```

Comment:
In format, is it valid to add 's0'? I think it should be 
format("Y-m-d H:i:s")